Build Error: no matching function for call to setattr(torch::jit::Node*&, ...)

Hello,
I'm trying to build pytorch (I tried both v3_buildfix and master branch) on a raspberry pi 3 but i'm getting that error:

```
torch/csrc/autograd/generated/VariableType.cpp: In member function ‘virtual std::tuple<at::Tensor, at::Tensor> torch::autograd::VariableType::prelu_backward(const at::Tensor&, const at::Tensor&, const at::Tensor&, std::array<bool, 2u>) const’:
torch/csrc/autograd/generated/VariableType.cpp:6114:65: error: no matching function for call to ‘setattr(torch::jit::Node*&, torch::jit::Symbol, std::array<bool, 2u>&)’
       setattr(n, jit::stringToSymbol("output_mask"), output_mask);
                                                                 ^
torch/csrc/autograd/generated/VariableType.cpp:32:13: note: candidate: void torch::autograd::setattr(torch::jit::Node*, torch::jit::Symbol, int64_t)
 static void setattr(jit::Node* n, jit::Symbol name, int64_t v)            
             ^
torch/csrc/autograd/generated/VariableType.cpp:32:13: note:   no known conversion for argument 3 from ‘std::array<bool, 2u>’ to ‘int64_t {aka long long int}’
torch/csrc/autograd/generated/VariableType.cpp:33:13: note: candidate: void torch::autograd::setattr(torch::jit::Node*, torch::jit::Symbol, const at::Scalar&)
 static void setattr(jit::Node* n, jit::Symbol name, const at::Scalar& v)  
             ^
torch/csrc/autograd/generated/VariableType.cpp:33:13: note:   no known conversion for argument 3 from ‘std::array<bool, 2u>’ to ‘const at::Scalar&’
torch/csrc/autograd/generated/VariableType.cpp:34:13: note: candidate: void torch::autograd::setattr(torch::jit::Node*, torch::jit::Symbol, const IntList&)
 static void setattr(jit::Node* n, jit::Symbol name, const at::IntList& v) 
             ^
torch/csrc/autograd/generated/VariableType.cpp:34:13: note:   no known conversion for argument 3 from ‘std::array<bool, 2u>’ to ‘const IntList& {aka const at::ArrayRef<long long int>&}’
torch/csrc/autograd/generated/VariableType.cpp:35:13: note: candidate: void torch::autograd::setattr(torch::jit::Node*, torch::jit::Symbol, bool)
 static void setattr(jit::Node* n, jit::Symbol name, bool v)               
             ^
torch/csrc/autograd/generated/VariableType.cpp:35:13: note:   no known conversion for argument 3 from ‘std::array<bool, 2u>’ to ‘bool’
torch/csrc/autograd/generated/VariableType.cpp:36:13: note: candidate: void torch::autograd::setattr(torch::jit::Node*, torch::jit::Symbol, double)
 static void setattr(jit::Node* n, jit::Symbol name, double v)             
             ^
torch/csrc/autograd/generated/VariableType.cpp:36:13: note:   no known conversion for argument 3 from ‘std::array<bool, 2u>’ to ‘double’
torch/csrc/autograd/generated/VariableType.cpp:38:13: note: candidate: template<long unsigned int N> void torch::autograd::setattr(torch::jit::Node*, torch::jit::Symbol, std::array<bool, N>)
 static void setattr(jit::Node* n, jit::Symbol name, std::array<bool, N> v)
             ^
torch/csrc/autograd/generated/VariableType.cpp:38:13: note:   template argument deduction/substitution failed:
torch/csrc/autograd/generated/VariableType.cpp:6114:65: note:   mismatched types ‘long unsigned int’ and ‘unsigned int’
       setattr(n, jit::stringToSymbol("output_mask"), output_mask);


error: command 'arm-linux-gnueabihf-gcc' failed with exit status 1arm-linux-gnueabihf-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/home/enstar/pytorch -I/home/enstar/pytorch/torch/csrc -I/home/enstar/pytorch/torch/lib/pybind11/include -I/home/enstar/pytorch/torch/lib/tmp_install/include -I/home/enstar/pytorch/torch/lib/tmp_install/include/TH -I/home/enstar/pytorch/torch/lib/tmp_install/include/THNN -I/home/enstar/pytorch/torch/lib/tmp_install/include/ATen -I/home/enstar/.local/lib/python2.7/site-packages/numpy/core/include -I/usr/include/python2.7 -c torch/csrc/autograd/functions/convolution.cpp -o build/temp.linux-armv7l-2.7/torch/csrc/autograd/functions/convolution.o -D_THP_CORE -std=c++11 -Wno-write-strings -fno-strict-aliasing -Wno-missing-braces -DWITH_NUMPY

```
Note: I'm building without CUDA and DISTRIBUTED support.

OS:  Ubuntu 16.04 (Mate)
gcc: arm-linux-gnueabihf-gcc (Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.5) 5.4.0 20160609
cmake:  v3.5.1

--- a/tools/autograd/templates/VariableType.cpp
+++ b/tools/autograd/templates/VariableType.cpp
@@ -15,6 +15,7 @@
 #include <initializer_list>
 #include <iostream>
 #include <functional>
+#include <cstddef>
 
 #ifdef _MSC_VER
 #ifdef Type
@@ -37,7 +38,7 @@ static void setattr(jit::Node* n, jit::Symbol name, SparseTensor s)        { n->
 static void setattr(jit::Node* n, jit::Symbol name, const at::IntList& v)  { n->is_(name, v); }
 static void setattr(jit::Node* n, jit::Symbol name, bool v)                { n->i_(name, v); }
 static void setattr(jit::Node* n, jit::Symbol name, double v)              { n->f_(name, v); }
-template<unsigned long N>
+template<std::size_t N>
 static void setattr(jit::Node* n, jit::Symbol name, std::array<bool, N> v) { n->is_(name, std::vector<int64_t>(v.begin(), v.end())); }
 
 VariableType::VariableType(Context* context, Type* baseType)

